src/app/vc-receipt-status/middleware.ts

```diff
@@ -6,40 +6,55 @@ import { isAxiosError } from "axios";
 import config from "../../config/config";
 import { logger } from "../../lib/logger";
 
-enum AppVcReceiptStatus {
+export enum AppVcReceiptStatus {
   COMPLETED = "COMPLETED",
   ERROR = "ERROR",
   PROCESSING = "PROCESSING",
   INTERVENTION = "INTERVENTION",
 }
 
-export const getAppVcReceiptStatus: RequestHandler = async (
+export const getAppVcReceiptStatus = async (
   req: Request,
-  res: Response,
-) => {
+): Promise<AppVcReceiptStatus> => {
   try {
-    // For browser tests
-    if (config.ENABLE_PREVIEW && process.env.NODE_ENV === "local") {
-      res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
-      return;
+    // If we already have the processed journey in the session, return it.
+    if (
+      req.session?.journey &&
+      req.session.journey !== AppVcReceiptStatus.PROCESSING
+    ) {
+      return req.session.journey as AppVcReceiptStatus;
     }
-
     const appVcResponse = await appVcReceived(req);
     if (!isJourneyResponse(appVcResponse.data)) {
       throw new Error(
         "Journey response expected from successful check app vc receipt response.",
       );
     }
-
     req.session.journey = appVcResponse.data.journey;
-    res.status(200).json({ status: AppVcReceiptStatus.COMPLETED });
-    return;
+
+    return AppVcReceiptStatus.COMPLETED;
   } catch (error) {
     if (isAxiosError(error) && error.response?.status === 404) {
-      res.status(200).json({ status: AppVcReceiptStatus.PROCESSING });
-      return;
+      return AppVcReceiptStatus.PROCESSING;
     }
     logger.error(error, "Error getting app vc receipt status");
-    res.status(500).json({ status: AppVcReceiptStatus.ERROR });
+    return AppVcReceiptStatus.ERROR;
   }
 };
+
+export const getAppVcReceiptStatusAndStoreJourneyResponse: RequestHandler =
```

Comment:
I was thinking the above method would be getAppVcReceiptStatusAndStoreJourneyResponse (since it's actually storing the response) and this could be something like pollVcReceiptStatus ?